src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1493,9 +1494,18 @@ private void computeChangeLog(GitClient git, Revision revToBuild, TaskListener l
             }
 
             if (!exclusion) {
-                // this is the first time we are building this branch, so there's no base line to compare against.
-                // if we force the changelog, it'll contain all the changes in the repo, which is not what we want.
-                listener.getLogger().println("First time build. Skipping changelog.");
+                FirstBuildChangelog firstBuildChangelog = getExtensions().get(FirstBuildChangelog.class);
+                if (firstBuildChangelog!=null&&firstBuildChangelog.isMakeChangelog()) {
+                    changelog.to(out).max(1).execute();
+                    executed = true;
+                    listener.getLogger().println("First time build. Latest changes added to changelog.");
+                } else {
+                    // this is the first time we are building this branch, so there's no base line
+                    // to compare against.
+                    // if we force the changelog, it'll contain all the changes in the repo, which
+                    // is not what we want.
```

Comment:
I think this existing comment should be rewritten now that its within the context of the new option. But, perhaps just removing it would be best.

Reply:
Removed in 61bb7582896bae3d9c4863de6b76fa9a18f9a314